server.go

```diff
@@ -49,24 +55,47 @@ func startMonitoring(license string) {
 	}
 }
 
-func startApp(port string, env *cfenv.App) {
-	// Look for env vars first in a user provided service (if available), and if not found,
-	// fallback to the environment.
-	envVars := helpers.NewEnvVarsFromPath(
-		helpers.NewEnvLookupFromCFAppNamedService(env, cfUserProvidedService),
-		os.LookupEnv,
-	)
+func startApp(port string, app *cfenv.App) {
+	var opts []env.VarSetOpt
+
+	upsNames := os.Getenv(envUPSNames)
+
+	switch {
+	// Look for UPS names in a delimited environment variable.
+	case upsNames != "":
+		opts = append(opts, env.WithOSLookup())
+
+		// TODO(jonathaningram): we already have an app. Is this so we're making
```

Comment:
you don't need to re-grab the cfenv.Current() since we already pass it in. just check if it's nil. don't use a switch-case in this instance.

```go
if upsNames != "" && app != nil {
  // create UPS Opts
} else {
  // create default opts
}
```

also, move the logic for each case into functions. makes it easier to read.